go/libraries/doltcore/env/actions/diffs.go

```diff
@@ -104,3 +120,66 @@ func GetTableDiffs(ctx context.Context, dEnv *env.DoltEnv) (*TableDiffs, *TableD
 
 	return stagedDiffs, notStagedDiffs, nil
 }
+
+func NewDocDiffs(ctx context.Context, older *doltdb.RootValue, docDetails []*doltdb.DocDetails) (*DocDiffs, error) {
+	added, modified, removed, err := older.DocDiff(ctx, docDetails)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var docs []string
+	docs = append(docs, added...)
+	docs = append(docs, modified...)
+	docs = append(docs, removed...)
+	sort.Strings(docs)
+
+	docsToType := make(map[string]DocDiffType)
+	for _, nt := range added {
+		docsToType[nt] = AddedDoc
+	}
+
+	for _, nt := range modified {
+		docsToType[nt] = ModifiedDoc
+	}
+
+	for _, nt := range removed {
+		docsToType[nt] = RemovedDoc
+	}
+
+	return &DocDiffs{len(added), len(modified), len(removed), docsToType, docs}, err
+}
+
+func (nd *DocDiffs) Len() int {
+	return len(nd.Docs)
+}
+
+func GetDocDiffs(ctx context.Context, dEnv *env.DoltEnv) (*DocDiffs, error) {
+	workingRoot, err := dEnv.WorkingRoot(ctx)
+
+	if err != nil {
+		return nil, RootValueUnreadable{WorkingRoot, err}
+	}
+
+	licenseText, err := dEnv.GetLocalFileText(env.LicenseFile)
+
+	if err != nil {
+		return nil, err
+	}
+	readmeText, err := dEnv.GetLocalFileText(env.ReadmeFile)
+	if err != nil {
+		return nil, err
+	}
+
+	docDetails := make([]*doltdb.DocDetails, 2)
```

Comment:
if you get rid of the make function you can define this as:

    docDetails := []*doltdb.DocDetails {
        &doltdb.DocDetails{licenseText, doltdb.LicensePk, nil},
        &doltdb.DocDetails{readmeText, doltdb.ReadmePk, nil}
    }